### PR TITLE
Add layer_id to each well connection

### LINKED ISSUE #277: Add layer_id to each well connection
If the user specifies multiple vertical reservoir zones ("layers") each well connection should be associated to a layer.

The resulting layer_id doesn't need to be used for anything yet - but we should try to write a test that things work.

### PR DESCRIPTION
**Pull request by LonnekevB**
*Insert a description of your pull request (PR) here, and check off the boxes below when they are done.*

---

### Contributor checklist

- [ ] :tada: This PR closes #277.
- [ ] :scroll: I have broken down my PR into the following tasks:
   - [ ] Task 1
   - [ ] Task 2
- [ ] :robot: I have added tests, or extended existing tests, to cover any new features or bugs fixed in this PR.
- [ ] :book: I have considered adding a new entry in `CHANGELOG.md`.
- [ ] :books: I have considered updating the documentation.

### CODE REVIEW

**Comment by LonnekevB** [comment]
New issue has been made on adapting the perforation strategy methods to layering. Not sure whether to make unit tests for the well_connections function within this pull request or within the new pull request on layering in perforation strategy (as these methods are kind of incorporated in the well_connections method).

File: src/flownet/data/from_flow.py [hunk 1]
```diff
@@ -39,25 +42,39 @@ def __init__(
...
         configured perforation strategy.
 
         Returns:
-            columns: WELL_NAME, X, Y, Z, DATE, OPEN
+            columns: WELL_NAME, X, Y, Z, DATE, OPEN, LAYER_ID
 
         """
+        if len(self._layers) > 0 and self._grid.nz is not self._layers[-1][-1]:
+            raise ValueError(
+                f"Number of layers from config ({self._layers[-1][-1]}) is not equel to "
```
Comment (wouterjdb) [inline src/flownet/data/from_flow.py:62]:
```suggestion
                f"Number of layers from config ({self._layers[-1][-1]}) is not equal to "
```

**Review by wouterjdb — CHANGES_REQUESTED** [review]
> New issue has been made on adapting the perforation strategy methods to layering. 

Good. For reference: #290 

> Not sure whether to make unit tests for the well_connections function within this pull request or within the new pull request on layering in perforation strategy (as these methods are kind of incorporated in the well_connections method).

You can do that in the new PR.

This looks very good, some minor comments and we're ready to go 👍

**Review by LonnekevB — COMMENTED** [review]
All changes are made

**Review by wouterjdb — APPROVED** [review]
LGTM 👍

**Pull request merged**